File: FSharpConsoleApp/Equality.fs
module FSharpConsoleApp.Equality

type Data = { Name : string ; Values : string[] }

let compute () =
    let a =
        {
           Name = "Charles"
           Values = [| "1" ; "2" |]
        }

    let b =
        {
           Name = "Charles"
           Values = [| "1" ; "2" |]
        }

    a = b

File: CSharpConsoleApp/Equality.cs
namespace CSharpConsoleApp
{
    public class Equality
    {
        public record Data(string Name, string[] Values);

        public bool Compute()
        {
            var a = new Data(
                "Charles",
                new[] {"1", "2"});

            var b = new Data(
                "Charles",
                new[] {"1", "2"});

            return a == b;
        }
    }
}